src/components/TodoAppComponent/TodoAppMain/TodoList.tsx

```diff
@@ -0,0 +1,144 @@
+import cn from 'classnames';
+import {
+  FormEvent, KeyboardEvent, useEffect, useRef, useState,
+} from 'react';
+import { Todo } from '../../../types/Todo';
+import { useTodosContext } from '../../../Context/TodosContext';
+
+export const TodoList = () => {
+  const [isEditById, setIsEditById] = useState<number | null>(null);
+  const [tempTitle, setTempTitle] = useState('');
+  const todoInputRef = useRef<HTMLInputElement>(null);
+
+  const {
+    tempTodo, loading, setLoading, handleDeleteTodo,
+    handleClickCheck, editTitle, filteredTodos,
+  } = useTodosContext();
+
+  const handleDoubleClickEdit = (todo: Todo) => {
+    const { id, title } = todo;
+
+    setIsEditById(id);
+    setTempTitle(title);
+  };
+
+  useEffect(() => {
+    if (todoInputRef.current) {
+      if (isEditById) {
+        todoInputRef.current.focus();
+      } else {
+        todoInputRef.current.blur();
+      }
+    }
+  }, [isEditById]);
+
+  const handleEditTitle = (event: FormEvent) => {
+    setLoading(true);
+    event.preventDefault();
+    if (isEditById !== null) {
```

Comment:
```suggestion
    if (isEditById) {
```

Reply:
https://prnt.sc/XC38mkbs64GZ - type problem

Reply:
okay so lets leave it as it was 👀